Add a specification for robot flags

Adimote opened the pull request:
We decided we don't want robot badges this year, but we still
want robots to have their corner identifiable, so we decided
robots should have a flag mount.

The reasoning for flags instead of badges is they require a smaller footprint and can be retro-fit to a robot more-so than badges. Now that it's 2 weeks before the competition, flags are more likely to be put onto a robot.

Adimote committed:
Add a specification for robot flags

We decided we don't want robot badges this year, but we still
want robots to have their corner identifiable, so we decided
robots should have a flag mount.

RealOrangeOne commented:
Those dimensions feel absolutely massive for the flag (and the pole for that matter)

I'd also suggest explicitly stating it doesnt count towards their dimensions (unless it does, in which case it really shouldnt)

Adimote commented:
We need to remember that robots are up to 50cm big, so a 50cm high flag isn't too big. Good point on the size definition

RealOrangeOne commented:
It'd be deal if all they needed was mounting for the flags (ideally velcro) so it's easier for them to design / build, and us to check sizing. It'll also make the flags look far less terrible, especially if we're planning to make them out of wire, which is far too flexible for this.

I still think the flag is _far_ too large, and overcomplicated. I think using something simple like a coloured panel of wood mounted to the top of the robot will easily do.

Adimote commented:
Hmm, what if we give them like a stack of 10  10cm squares, and then tell them they need velcro spaces for at least 2, then they can have fun with incorporating the colouring into their robots design?

RealOrangeOne commented:
ooh, i like that! Although 10cm might be a bit much? What about 5cm, and at least 3, then theoretically no matter what angle we're looking at, we'll see at least 1?

PeterJCLaw commented on fig-flag.svg:
```diff
@@ -0,0 +1,136 @@
+         id="tspan827-7-27"
+         x="181.94379"
+         y="50.795963"
+         style="font-style:normal;font-variant:normal;font-weight:bold;font-stretch:normal;font-size:19.75555611px;font-family:Sans-serif;-inkscape-font-specification:'Sans-serif Bold';stroke-width:0.26458332px">40cm</tspan></text>
```
We should be consistent about our units -- the text describing the spec uses mm, so the diagram should too.

Adimote replied:
fixed in 96616922a530f79b2f0f256625f4c4a48c1a6227

Adimote replied:
fixed in 96616922a530f79b2f0f256625f4c4a48c1a6227

PeterJCLaw commented on specs.tex:
```diff
@@ -66,6 +66,11 @@ \subsection{Arena}
         that of other zones.
   \item Tokens will be placed in the scoring zone on the opposite side to their
         matching coloured scoring zone.
+  \item \label{spec:flags} Flags will be made from wire folded in half
```
why is the folding in half relevant? I'm not sure I understand what it is that we're trying to describe with this.

PeterJCLaw replied:
* What restrictions are there about the flag and how the flag is mounted?
* Could the pole be horizontal?
* Can the flag be an April Tag Marker?
* Does the flag need to be at some minimum (or maximum) height?
* Does the flag need to be within the 50cm bounding box of the robot at the start of the match?

PeterJCLaw commented on specs.tex:
```diff
@@ -66,6 +66,11 @@ \subsection{Arena}
+  \item \label{spec:flags} Flags will be made from wire folded in half
+        to form a single rod with a maximum width of less than \si{10}{mm},
+        and length of more than \si{500}{mm}. The length of rod without
+        a flag will be \si{400}{mm}, as described in \figref{fig:flag}.
```
The phrase "The length of rod without a flag [is]" is not clear. I _think_ you mean "The length of rod not covered by the flag [is]", but it reads to me as "When the flag is not present, the length [is]".

PeterJCLaw replied:
Why do the rules care about the length of the wire, etc.?

Maybe this will be clearer if we specify which parts of this the robots need to provide and which parts we're providing.

PeterJCLaw reviewed (dismissed):
Please could you edit the description of the PR to include the reasons that you think flags are better than badges?

Historically, getting students to build flags was a complete failure; I'm not sure why we think we'll be more successful this time around.

~~

I'm not at all clear from the rules who is expected to provide which parts of the flag/pole/mount. If we're providing some part of it, we should be explicit about that. We should also focus on providing detail around the interface between what we provide and what the competitors provide. We _don't_ need to provide many details about the parts we're providing that the robots don't need to worry about (since doing so makes it harder for us to change things later).

Adimote committed:
change flag dimensions, mention they are PVC pipes

Adimote committed:
Mention flags are removable and should be upright

Also mention they should be held firmly in their mountings

Adimote committed:
dramatically downsize the flags

Adimote committed:
mention flags are providd by SourceBots

Adimote committed:
Mention flags don't count towards the robot size.

Adimote committed:
Require the flags to be visible

(We deliberately don't require the orientation fo the flag, as
it doesn't matter)

Adimote committed:
don't need to say they're PVC pipes

Adimote committed:
change flag diagram to be mm

Adimote commented:
@PeterJCLaw This time we're planning on making the flags for the teams. We will only expect them to have a mount for them. Aside from these rules we also have a great plan for mounting these flags: We have some pipe fittings which grip the flags very solidly. If we give them out to teams, and let them glue them to their robots, I'm confident we'll be able to get 100% of all teams that haven't even read this update to retrofit flag mounts.

We are deliberately not specifying that flags don't need to face upwards, to make it easier for them to find a way of mounting their flags even on the competition day.

PeterJCLaw commented:
@Adimote I'll be able to have a better look at this tomorrow evening, though I'm a little concerned about the decision not to require that flags are upright. Some things I would consider when describing the mounting:
 * the interaction between the flags and access to the power button; a potential safety concern
 * that they might fall out if not upright; potentially also pulling out the competition mode usb stick and/or damaging the flag and/or usb stick and/or hub/Pi it's plugged into
 * ease of placing in a flag; given that this will happen before/after a match under time pressure it needs to be easy

Adimote committed:
add flags to changelog

Adimote committed:
s/of/with
